Build failure in ec_stdint.h/wdg.c on Alpine with GCC 8.3

I have been attempting to build ettercap at commit ffeaac725d31f10b270cd3ece14e5a7b83a1f1ff on Alpine but have experienced the following issue:

```
# make
Scanning dependencies of target ec_interfaces
[  1%] Building C object src/interfaces/CMakeFiles/ec_interfaces.dir/__/ec_interfaces.c.o
[  1%] Building C object src/interfaces/CMakeFiles/ec_interfaces.dir/text/ec_text.c.o
[  1%] Building C object src/interfaces/CMakeFiles/ec_interfaces.dir/text/ec_text_conn.c.o
[  2%] Building C object src/interfaces/CMakeFiles/ec_interfaces.dir/text/ec_text_display.c.o
[  2%] Building C object src/interfaces/CMakeFiles/ec_interfaces.dir/text/ec_text_plugin.c.o
[  3%] Building C object src/interfaces/CMakeFiles/ec_interfaces.dir/text/ec_text_profile.c.o
[  3%] Building C object src/interfaces/CMakeFiles/ec_interfaces.dir/daemon/ec_daemon.c.o
[  3%] Building C object src/interfaces/CMakeFiles/ec_interfaces.dir/curses/widgets/wdg.c.o
In file included from /ettercap/e/src/interfaces/curses/widgets/wdg.h:21,
                 from /ettercap/e/src/interfaces/curses/widgets/wdg.c:22:
/ettercap/e/include/ec_stdint.h:37:10: error: unknown type name 'u_int8_t'
  typedef u_int8_t   u_int8;
          ^~~~~~~~
<snip>

In file included from /ettercap/e/src/interfaces/curses/widgets/wdg.c:22:
/ettercap/e/src/interfaces/curses/widgets/wdg.h:161:4: error: unknown type name 'u_char'
    u_char screen_color;

<output truncated>
```

I do not experience the same issue on Ubuntu 18.04.

However, upon further investigation, it appears that neither `ec_stdint.h` or `wdg.h` explicitly import `sys/types.h` for Linux (`wdg.h` does for Darwin) which is where these types are defined. 

On Ubuntu 18.04, it appears that `sys/types.h` is being implicitly imported within `wdg.h` when it imports `stdlib.h`. I.e.

```bash
$ echo "#include <stdlib.h>" | gcc -E - | grep -e "\(u_int8_t\)\|\(u_char\)"
typedef unsigned char __u_char;
typedef __u_char u_char;
typedef unsigned int u_int8_t __attribute__ ((__mode__ (__QI__)));
```

However, on Alpine this is not the case. I did not trace the imports to see the difference.

It appears there are two possible solutions, the first is to tweak `ec_stdint.h` to unconditionally include `sys/types.h` on Linux (perhaps for other OS'es as well?). That is, by adding:

```C
#if defined(OS_LINUX)
        #include <sys/types.h>
#endif
```

Alternatively, `wdg.h` can be updated as follows:
```diff
--- a/src/interfaces/curses/widgets/wdg.h
+++ b/src/interfaces/curses/widgets/wdg.h
@@ -14,7 +14,7 @@
    #include <windows.h>
 #endif
 
-#ifdef OS_DARWIN
+#if defined(OS_DARWIN) || defined(OS_LINUX)
    #include <sys/types.h>
 #endif
```

I am inclined to think the first is the better solution, however, the second appears to be more aligned with the existing code base. If needed, I can submit a PR.

--- a/include/ec_stdint.h
+++ b/include/ec_stdint.h
@@ -34,10 +34,10 @@
 	typedef int32_t   int32;
 	typedef int64_t   int64;
 
-	typedef u_int8_t   u_int8;
-	typedef u_int16_t  u_int16;
-	typedef u_int32_t  u_int32;
-	typedef u_int64_t  u_int64;
+	typedef uint8_t   u_int8;
+	typedef uint16_t  u_int16;
+	typedef uint32_t  u_int32;
+	typedef uint64_t  u_int64;
 
    #if defined OS_BSD_OPEN && !defined HAVE_STDINT_H
       #define INT8_MAX     CHAR_MAX

--- a/include/ec_stdint.h
+++ b/include/ec_stdint.h
@@ -4,40 +4,40 @@
 #include <limits.h>
 
 #if defined HAVE_STDINT_H && !defined OS_SOLARIS
-	#include <stdint.h>
+   #include <stdint.h>
 #elif !defined OS_SOLARIS
-	#include <sys/types.h>
+   #include <sys/types.h>
 #elif defined OS_SOLARIS
-	#include <sys/inttypes.h>
+   #include <sys/inttypes.h>
 #endif
 
 #if defined HAVE_INTTYPES_H
-	#include <inttypes.h>
+   #include <inttypes.h>
 #else /* HAVE_INTTYPES_H */
-	#if __WORDSIZE == 64
-		#define __PRI64_PREFIX "l"
-	#else
-		#define __PRI64_PREFIX "ll"
-	#endif
+   #if __WORDSIZE == 64
+      #define __PRI64_PREFIX "l"
+   #else
+      #define __PRI64_PREFIX "ll"
+   #endif
 #ifndef PRId64
-	#define PRId64 __PRI64_PREFIX "d"
+   #define PRId64 __PRI64_PREFIX "d"
 #endif
 #ifndef PRIu64
-	#define PRIu64 __PRI64_PREFIX "u"
+   #define PRIu64 __PRI64_PREFIX "u"
 #endif
 #endif /* HAVE_INTTYPES_H */
 
 #ifndef TYPES_DEFINED
 #define TYPES_DEFINED
-	typedef int8_t    int8;
-	typedef int16_t   int16;
-	typedef int32_t   int32;
-	typedef int64_t   int64;
+   typedef int8_t    int8;
+   typedef int16_t   int16;
+   typedef int32_t   int32;
+   typedef int64_t   int64;
 
-	typedef uint8_t   u_int8;
-	typedef uint16_t  u_int16;
-	typedef uint32_t  u_int32;
-	typedef uint64_t  u_int64;
+   typedef uint8_t   u_int8;
+   typedef uint16_t  u_int16;
+   typedef uint32_t  u_int32;
+   typedef uint64_t  u_int64;
 
    typedef unsigned char u_char;
 
@@ -52,24 +52,24 @@
 
    /* Maximum of signed integral types.  */
    #ifndef INT8_MAX
-      #define INT8_MAX		(127)
+      #define INT8_MAX     (127)
    #endif
    #ifndef INT16_MAX
-      #define INT16_MAX		(32767)
+      #define INT16_MAX    (32767)
    #endif
    #ifndef INT32_MAX
-      #define INT32_MAX		(2147483647)
+      #define INT32_MAX    (2147483647)
    #endif
 
    /* Maximum of unsigned integral types.  */
    #ifndef UINT8_MAX
-      #define UINT8_MAX		(255)
+      #define UINT8_MAX    (255)
    #endif
    #ifndef UINT16_MAX
-      #define UINT16_MAX	(65535)
+      #define UINT16_MAX   (65535)
    #endif
    #ifndef UINT32_MAX
-      #define UINT32_MAX	(4294967295U)
+      #define UINT32_MAX   (4294967295U)
    #endif
    
 #endif

--- a/include/ec_stdint.h
+++ b/include/ec_stdint.h
@@ -39,6 +39,8 @@
 	typedef uint32_t  u_int32;
 	typedef uint64_t  u_int64;
 
+   typedef unsigned char u_char;
+
    #if defined OS_BSD_OPEN && !defined HAVE_STDINT_H
       #define INT8_MAX     CHAR_MAX
       #define UINT8_MAX    UCHAR_MAX

